cmd/scollector/collectors/riak.go

```diff
@@ -0,0 +1,506 @@
+		RateType: metadata.Gauge,
+		Unit:     metadata.Count,
+		Desc:     "99th percentile of siblings encountered during all GET operations by this node within the last minute.",
+	},
+	"node_get_fsm_siblings_100": {
+		Metric:   "node.siblings.100th",
+		TagSet:   opentsdb.TagSet{"type": "get"},
+		RateType: metadata.Gauge,
+		Unit:     metadata.Count,
+		Desc:     "100th percentile of siblings encountered during all GET operations by this node within the last minute.",
+	},
+	"node_get_fsm_rejected_total": {
+		Metric:   "node.requests.rejected",
+		TagSet:   opentsdb.TagSet{"type": "get"},
+		RateType: metadata.Counter,
+		Unit:     metadata.Event,
+		Desc:     "Total number of GET FSMs rejected by Sidejob's overload protection.",
+	},
+	"node_put_fsm_rejected_total": {
+		Metric:   "node.requests.rejected",
+		TagSet:   opentsdb.TagSet{"type": "put"},
+		RateType: metadata.Counter,
+		Unit:     metadata.Event,
+		Desc:     "Total number of PUT FSMs rejected by Sidejob's overload protection.",
+	},
+	"ring_num_partitions": {
+		Metric:   "ring_num_partitions",
+		RateType: metadata.Gauge,
+		Unit:     metadata.Count,
+		Desc:     "The number of partitions in the ring.",
+	},
+	"ring_creation_size": {
+		Metric:   "ring.creation_size",
+		RateType: metadata.Gauge,
+		Unit:     metadata.Count,
+		Desc:     "Ring size this cluster was created with.",
+	},
+	"cpu_nprocs": {
+		Metric:   "cpu.nprocs",
+		RateType: metadata.Gauge,
+		Unit:     metadata.Count,
+		Desc:     "Number of operating system processes.",
+	},
+	"cpu_avg1": {
+		Metric:   "cpu.avg1",
+		RateType: metadata.Gauge,
+		Unit:     metadata.Load,
+		Desc:     "The average number of active processes for the last 1 minute (equivalent to top(1) command’s load average when divided by 256()).",
+	},
+	"cpu_avg5": {
+		Metric:   "cpu.avg5",
+		RateType: metadata.Gauge,
+		Unit:     metadata.Load,
+		Desc:     "The average number of active processes for the last 5 minutes (equivalent to top(1) command’s load average when divided by 256()).",
+	},
+	"cpu_avg15": {
+		Metric:   "cpu.avg15",
+		RateType: metadata.Gauge,
+		Unit:     metadata.Load,
+		Desc:     "The average number of active processes for the last 15 minutes (equivalent to top(1) command’s load average when divided by 256()).",
+	},
+	"riak_search_vnodeq_total": {
+		Metric:   "search.vnodeq",
+		RateType: metadata.Counter,
+		Unit:     metadata.Event,
+		Desc:     "Total number of unprocessed messages all vnode message queues in the Riak Search subsystem have received on this node since it was started.",
+	},
+	"riak_search_vnodes_running": {
+		Metric:   "search.vnodes_running",
+		RateType: metadata.Gauge,
+		Unit:     metadata.Process,
+		Desc:     "Total number of vnodes currently running in the Riak Search subsystem.",
+	},
+}
+
+func init() {
+	collectors = append(collectors, &IntervalCollector{F: c_riak, Enable: enableRiak})
+}
+
+const (
+	riakURL string = "http://localhost:8098/stats"
```

Comment:
We assume localhost and default port in other collectors, so I think this
is fine for now
On Mar 19, 2015 4:45 AM, "Dmitry V. Medvedev" <notifications@github.com>
wrote:

> In cmd/scollector/collectors/riak.go
> <https://github.com/bosun-monitor/bosun/pull/801#discussion_r26736568>:
>
> > +		Desc:     "Total number of unprocessed messages all vnode message queues in the Riak Search subsystem have received on this node since it was started.",
> > +	},
> > +	"riak_search_vnodes_running": {
> > +		Metric:   "search.vnodes_running",
> > +		RateType: metadata.Gauge,
> > +		Unit:     metadata.Process,
> > +		Desc:     "Total number of vnodes currently running in the Riak Search subsystem.",
> > +	},
> > +}
> > +
> > +func init() {
> > +	collectors = append(collectors, &IntervalCollector{F: c_riak, Enable: enableRiak})
> > +}
> > +
> > +const (
> > +	riakURL string = "http://localhost:8098/stats"
>
> That can be set in riak.config . I'm not yet sure how to better implement
> that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bosun-monitor/bosun/pull/801/files#r26736568>.
>